mealie/routes/recipe/recipe_crud_routes.py

```diff
@@ -118,20 +119,22 @@
     @router_exports.get("/{slug}/exports/zip")
     def get_recipe_as_zip(self, slug: str, token: str, temp_path=Depends(temporary_zip_path)):
         """Get a Recipe and It's Original Image as a Zip File"""
-        slug = validate_recipe_token(token)
+        validated_slug = validate_recipe_token(token)
 
-        if slug != slug:
+        if validated_slug != slug:
             raise HTTPException(status_code=400, detail="Invalid Slug")
 
-        recipe: Recipe = self.mixins.get_one(slug)
+        recipe: Recipe = self.mixins.get_one(validated_slug)
         image_asset = recipe.image_dir.joinpath(RecipeImageTypes.original.value)
         with ZipFile(temp_path, "w") as myzip:
             myzip.writestr(f"{slug}.json", recipe.model_dump_json())
 
             if image_asset.is_file():
                 myzip.write(image_asset, arcname=image_asset.name)
 
-        return FileResponse(temp_path, filename=f"{slug}.zip")
+        return FileResponse(
+            temp_path, filename=f"{recipe.slug}.zip", background=BackgroundTask(temp_path.unlink, missing_ok=True)
```

Comment:
If anyone has an idea how to get CodeQL to pass, let me know.  `recipe.slug` is a cleaned/slugified value from the db.  I wouldn't really characterize that as user provided value.  The path isn't used to access any resource in the mealie container, but to provide an advisory filename to the client.  By definition (since it was slugified), it can't contain any characters that would allow accessing different directories.  

At this point the only option is to change the interface to return a uuid name instead of a slug name, and I don't feel comfortable doing that without input from someone more involved in the project.